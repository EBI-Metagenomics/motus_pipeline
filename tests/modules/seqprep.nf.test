nextflow_process {

    name "Test Process SEQPREP"
    script "modules/seqprep.nf"
    process "SEQPREP"

    test("Overlap on PE") {

        when {
            params {
                name = "example_paired"
                reads = "${baseDir}/tests/modules/fixtures/fastp/paired_end/example_paired_{1,2}.fastq.gz"
            }
            process {
                """
                input[0] = params.name
                input[1] = channel.fromFilePairs("${params.reads}", checkIfExists: true).map { it[1] }
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
