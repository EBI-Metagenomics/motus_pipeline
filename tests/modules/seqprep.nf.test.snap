{
    "Overlap on PE": {
        "content": [
            {
                "overlapped_reads": [
                    "example_paired_merged.fastq.gz:md5,57dff46d32b9a1f391dda1af7274fd3c"
                ],
                "forward_unmapped_reads": [
                    "example_paired_forward_unmerged.fastq.gz:md5,a8150869f2b9ee3f6ab78956363240ef"
                ],
                "reverse_unmerged_reads": [
                    "example_paired_reverse_unmerged.fastq.gz:md5,3f8f7b018e2154fb3a75c17cb817d9de"
                ]
            }
        ],
        "timestamp": "2023-03-30T09:12:48+0000"
    }
}